### ISSUE TITLE
Linebreaks of source code is removed from rendered HTML

### ISSUE THREAD

**Issue by msh-odoo** [odoo/owl#727]
I am not sure this is owl related issue or not but posting here to make sure.

When owl renders temlate, say for example template is something like:
```
<div class="a">
        <div class="b">
            <span>Hello World!</span>
        </div>
</div>
```

When HTML is rendered through owl when I edit HTML using the developer tool I found linebreaks are not preserved and the code looks like:
```
<div class="a"><div class="b"><span>Hello World!</span></div></div>
```

This usually don't create an issue but it creates issue when two elements inline, not that for line breaks browser adds space between two inline elements, as rendered code does not have linebreaks preserved so spaces are removed, we explicitly have to add padding between two inline elements.

**Comment by ged-odoo** [comment]
@msh-odoo This is an intentional behaviour.  The reason is that all those whitespaces/new lines add a huge amount of nodes to the final DOM, which in turn, add a large performance cost to the owl diff engine, because it has to traverse all these nodes all the time. And this is sad, because 99% of the time, the new line actually do not have any value, they just exist to make it easy to read the template.

So, what Owl does is documented here: https://github.com/odoo/owl/blob/master/doc/reference/qweb_templating_language.md#white-spaces

Basically, it will remove all non significant whitespaces, except if you are in a `<pre>` tag or if they are on the same line. So, in your case, you can just write your template in one line, with a space where you need it:

```xml
<div class="a"><div class="b"> <span>Hello World!</span></div></div>
```
or you can use css.

Does that solve your issue?

**Comment by msh-odoo** [comment]
@ged-odoo Thanks for your detailed reply, Intention behind this issue was to confirm it is intended behaviour or not, yes css can surely fix my issue, and yes those whitespace and linebreak create issue for typical case like in my case two inline elements which adds space.

I am closing this issue, Thanks.
